add all demo gif generation

tharunkumartk opened the pull request:
The `generate_demo_video.py` script now supports automatic discovery and batch generation of demonstration videos. This PR contains all of the gifs generated from all of the existing demonstrations. This addresses issue #51 by adding an easy mechanism to regenerate all demo gifs.

Also updated the environment generation to match the new demo gif directory structure. 


### 1. Latest Demo Generation
Generate a GIF for the most recently collected demonstration:
```bash
python scripts/generate_demo_video.py --latest
```

This will:
- Find the most recent `.p` file across all environments
- Generate a GIF with descriptive filename: `latest_{env}_{seed}_{timestamp}.gif`
- Save to `docs/envs/assets/demo_gifs/{env}/`

### 2. Batch Generation for All Demos
Generate GIFs for all collected demonstrations:
```bash
python scripts/generate_demo_video.py --all
```

Optional: Limit the number of demos processed:
```bash
python scripts/generate_demo_video.py --all --max-demos 10
```

This will:
- Discover all `.p` files in the demos directory
- Process demos in chronological order (newest first)
- Generate GIFs with unique filenames: `{env}_seed{seed}_{timestamp}.gif`
- Organize output by environment in subdirectories
- Show progress and summary statistics

### 3. Environment-Specific Generation
Generate GIFs for all demos of a specific environment:
```bash
python scripts/generate_demo_video.py --env prbench/Motion2D-p1
```

Or with the sanitized environment name:
```bash
python scripts/generate_demo_video.py --env Motion2D-p1
```

### 4. Original Single Demo Mode (Preserved)
The original functionality still works:
```bash
python scripts/generate_demo_video.py demos/Motion2D-p1/0/1752189500.p
```

## Common Options

All modes support these options:

- `--demos-dir PATH`: Specify demos directory (default: `demos`)
- `--output-dir PATH`: Specify output directory (default: `docs/envs/assets/demo_gifs/`)
- `--fps N`: Set frames per second (default: from environment metadata)
- `--loop N`: Set GIF loop count (0 = infinite, default: 0)
- `--max-demos N`: Limit demos processed (for `--all` and `--env` modes)

## Directory Structure

The script expects demos to be organized as:
```
demos/
├── ENV_ID/
│   ├── SEED/
│   │   ├── TIMESTAMP.p
│   │   └── ...
│   └── ...
└── ...
```

Generated GIFs are now organized by environment:
```
docs/envs/assets/demo_gifs/
├── Motion2D-p1/
│   ├── latest_Motion2D-p1_seed0_1752189500.gif
│   ├── Motion2D-p1_seed1_1757990591.gif
│   ├── Motion2D-p1_seed2_1757990595.gif
│   └── ...
├── ClutteredRetrieval2D-o1/
│   ├── ClutteredRetrieval2D-o1_seed2_1752266113.gif
│   └── ...
├── DynObstruction2D-o2/
│   ├── DynObstruction2D-o2_seed0_1757345215.gif
│   ├── DynObstruction2D-o2_seed0_1757556526.gif
│   └── ...
└── ...
```


## Examples

```bash
# Generate GIF for latest demo
python scripts/generate_demo_video.py --latest

# Generate GIFs for all demos (up to 5)
python scripts/generate_demo_video.py --all --max-demos 5

# Generate GIFs for Motion2D-p1 environment only
python scripts/generate_demo_video.py --env prbench/Motion2D-p1

# Use custom output directory
python scripts/generate_demo_video.py --latest --output-dir my_gifs/

# Original single file mode
python scripts/generate_demo_video.py demos/Motion2D-p1/0/1752189500.p
```

## Example Output

When running batch processing, you'll see output like this:

```
Found 90 demos. Generating GIFs...
[1/90] Skipping demos/Motion2D-p1/8/1757990616.p (GIF exists)
[2/90] Processing demos/Motion2D-p1/7/1757990612.p
Loaded demo for environment: prbench/Motion2D-p1
Demo length: 45 actions
Demo seed: 7
Saving video to docs/envs/assets/demo_gifs/Motion2D-p1/Motion2D-p1_seed7_1757990612.gif
Video specs: 46 frames, 20 fps
Video saved successfully!
[3/90] Error processing demos/Obstruction2D-o0/1/1751653925.p: Demo does not contain seed information.
[4/90] Processing demos/ClutteredRetrieval2D-o1/2/1752266113.p
...

Completed: 75 successful, 3 failed, 12 skipped

=== Failed Demos (3) ===
  • demos/Obstruction2D-o0/1/1751653925.p: Demo does not contain seed information.
  • demos/Motion2D-p2/1/corrupt_file.p: Error loading demo: Invalid pickle data
  • demos/StickButton2D-b1/0/1752262180.p: Not enough frames collected to create a video
```

yixuanhuang98 reviewed (approved):
Thanks, this PR LGTM! After you address the comments from @tomsilver, you could merge to the main branch.

tharunkumartk commented:
@tomsilver, I've updated the environment generation so that it now picks the first item in the new directory format. I removed everything from the previous directory format into the new directory format.
